forte/processors/base/data_augment_processor.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processors that augment the data.
+"""
+from abc import abstractmethod
+from forte.common.resources import Resources
+from forte.common.configuration import Config
+from forte.processors.data_augment.algorithms.dictionary_replacement_augmenter import DictionaryReplacementAugmenter
+from forte.processors.base.pack_processor import MultiPackProcessor
+__all__ = [
+    "BaseDataAugmentProcessor"
+]
+
+
+class BaseDataAugmentProcessor(MultiPackProcessor):
+    r"""The base class of processors that augment data.
+    This processor instantiates an augmenter where specific
+    data augmentation algorithms are embedded. The augmenter
+    will run the algorithms and the processor will pack the
+    strings.
+
+    The DA methods can all be considered as replacement-based
+    methods with different levels: character, word, sentence.
```

Comment:
I am a little not sure about whether they can "all be considered". Maybe we should not be so sure here. Btw, if you want to use the abbreviation form "DA", make sure you declare it before like this: Data Augmentation (DA)